File: src/Api.Core/Entities/Core/Event.cs
using Api.Core.Entities.Core.Interfaces;

namespace Api.Core.Entities.Core;

public class Event : IPrimaryEntity, ISearchableEntity
{
    public Guid Id { get; set; }
    public Guid CongregationId { get; set; }
    public Congregation Congregation { get; set; } = null!;
    public Guid OrganizationId { get; set; }
    public Organization Organization { get; set; } = null!;
    public Guid OrganizerId { get; set; }
    public Member Organizer { get; set; } = null!;
    public string Name { get; set; } = string.Empty;
    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }
    public TimeOnly? StartTime { get; set; }
    public TimeOnly? EndTime { get; set; }
    public string? Location { get; set; } = string.Empty;
    public int? Capacity { get; set; }
    public string? Description { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? DeletedAt { get; set; }
}

File: frontend/src/types/event.ts
export interface Event {
  id: string;
  organizationId: string;
  organizationName: string;
  organizerId: string;
  organizerName: string;
  name: string;
  startDate: string; // ISO date format YYYY-MM-DD
  endDate: string;   // ISO date format YYYY-MM-DD
  startTime?: string | null; // e.g. "09:00:00"
  endTime?: string | null;   // e.g. "11:30:00"
  location?: string | null;
  capacity?: number | null;
  description?: string | null;
  createdAt: string;
  registeredCount?: number;
  attendedCount?: number;
}

export interface Organization {
  id: string;
  name: string;
  description?: string | null;
  createdAt: string;
}

export interface EventRegistration {
  id: string;
  memberId: string;
  memberName: string;
  eventId: string;
  eventName: string;
  registrationDate: string;
  createdAt: string;
}

export interface EventAttendance {
  id: string;
  memberId: string;
  memberName: string;
  eventId: string;
  eventName: string;
  checkInTime: string; // e.g. "09:15:00"
  createdAt: string;
}

export interface EventFilters {
  name?: string;
  startDate?: string;
  endDate?: string;
  organizationId?: string;
  organizerId?: string;
}

export interface PagedResponse<T> {
  hasMore: boolean;
  nextCursor: string | null;
  data: T[];
}
